sdk/messaging/azservicebus/client.go

```diff
@@ -111,72 +116,123 @@ func newClientImpl(config clientConfig, options ...ClientOption) (*Client, error
 	return client, err
 }
 
-// NewProcessor creates a Processor, which allows you to receive messages from ServiceBus.
-func (client *Client) NewProcessor(options ...ProcessorOption) (*Processor, error) {
-	processor, err := newProcessor(client.namespace, options...)
+// NewProcessor creates a Processor for a queue.
+func (client *Client) NewProcessorForQueue(queue string, options ...ProcessorOption) (*Processor, error) {
+	options = append(options, ProcessorWithQueue(queue))
```

Comment:
Do we still need this option even though queue is a param now?

Reply:
I still have the option because, internally, it's still the easiest way for me to deal with it without having to expand to two functions everywhere. Might refactor later.

I did unexport it though, that was unintentional.